larsim/MCCheater/PhotonBackTracker.tcc

```diff
@@ -17,45 +17,52 @@ namespace cheat{
-      /*
-         if (fp.isValid()){
-         for( size_t btr_iter=0; btr_iter<priv_OpDetBTRs.size(); ++btr_iter){
-         auto btr=priv_OpDetBTRs.at(btr_iter);
-         auto od = btr->OpDetNum();
-         auto const& dr = fp.data(btr_iter);
-         for(auto& d : dr)
-         {
-         if(!d) continue;
-         priv_od_to_DivRec[od]=*d;//->ref();
-         }
+      for (auto& G4ModuleLabel : G4ModuleLabels) {
+        auto const& btrHandle = evt.template getValidHandle < std::vector < sim::OpDetBacktrackerRecord > > (G4ModuleLabel);
+        //      if(btrHandle.failedToGet()){
+        /*  mf::LogWarning("PhotonBackTracker") << "failed to get handle to     simb::MCParticle from "
+         *              << fG4ModuleLabel
+         *                          << ", return";*/ //This is now silent as it is expected to    happen every generation run. It is also temporary while we wait for
+        /*if( 0 ){ return;} //Insert check for DivRecs here, or don't use validHandle below.
+          auto const& divrecHandle = evt.template getValidHandle <std::vector<sim::OpDetDivRec>>(fWavLabel);
+          if(divrecHandle.failedToGet()){
+          return;
+          }*/
 
-         }
-         }else{throw cet::exception("PhotonBackTracker")<<"find Waveforms and DivRecs from BTRs failed.";}
-         */
+        art::fill_ptr_vector(priv_OpDetBTRs, btrHandle);
+        if (!std::is_sorted(priv_OpDetBTRs.begin(), priv_OpDetBTRs.end(), compareBTRlambda))
+          std::sort(priv_OpDetBTRs.begin(),priv_OpDetBTRs.end(),compareBTRlambda);
 
+        //art::fill_ptr_vector(priv_DivRecs, divrecHandle);
+        //auto compareDivReclambda = [](art::Ptr<sim::OpDetDivRec> a, art::Ptr<sim::OpDetDivRec> b) {return(a->OpDetNum() < b->OpDetNum());};
+        /*if (!std::is_sorted(priv_DivRecs.begin(), priv_DivRecs.end(), compareDivReclambda))
+          std::sort(priv_DivRecs.begin(), priv_DivRecs.end(), compareDivReclambda);*/
+        //art::FindManyP<raw::OpDetWaveform, sim::OpDetDivRec> fp(priv_OpDetBTRs, evt, fWavLabel);// fp;
+        //art::FindOneP<raw::OpDetWaveform, sim::OpDetDivRec> fp(priv_OpDetBTRs, evt, fWavLabel);// fp;
+        //They come in sorted by BTR. Now make an index matched vector of data_t sorted by BTR. No. I need easy, not efficient. Map of DetNum to data_t. data_t is then channel mapped.
+        /*
+          if (fp.isValid()){
+          for( size_t btr_iter=0; btr_iter<priv_OpDetBTRs.size(); ++btr_iter){
+          auto btr=priv_OpDetBTRs.at(btr_iter);
+          auto od = btr->OpDetNum();
+          auto const& dr = fp.data(btr_iter);
+          for(auto& d : dr)
+          {
+          if(!d) continue;
+          priv_od_to_DivRec[od]=*d;//->ref();
+          }
+
+          }
+          }else{throw cet::exception("PhotonBackTracker")<<"find Waveforms and DivRecs from BTRs failed.";}
+        */
+      }
```

Comment:
So, this DivRec code was commented out since it was DUNE-specific when this was moved to LArSim. A quick search in the dune repos doesn't show the equivalent showing up anywhere, so probably best to keep it here for now. Maybe add an explanatory comment saying this is DUNE-specific code which hasn't been migrated anywhere better yet?

Reply:
Ok, I added a comment about it.